mungegithub/mungers/submit-queue-batch_test.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"strconv"
+	"strings"
+	"testing"
+
+	githubapi "github.com/google/go-github/github"
+)
+
+func expectEqual(t *testing.T, msg string, have interface{}, want interface{}) {
+	if !reflect.DeepEqual(have, want) {
+		t.Errorf("bad %s: got %v, wanted %v",
+			msg, have, want)
+	}
+}
+
+type stringHandler string
+
+func (h stringHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintf(w, "%s", h)
+}
+
+func TestGetSuccessfulBatchJobs(t *testing.T) {
+	body := `[
+	{"type":"batch", "repo":"a", "refs":"1", "state":"success", "context":"$"},
+	{"type":"batch", "repo":"a", "refs":"2", "state":"success", "context":"!"},
+
+	{"type":"pr", "repo":"a", "refs":"1", "state":"success", "context":"$"},
+	{"type":"batch", "repo":"b", "refs":"1", "state":"success", "context":"$"},
+	{"type":"batch", "repo":"a", "refs":"1", "state":"fail", "context":"$"}
+	]`
+	serv := httptest.NewServer(stringHandler(body))
+	defer serv.Close()
+	jobs, err := getSuccessfulBatchJobs("a", serv.URL)
+	if err != nil {
+		t.Fatal(err)
+	}
+	expectEqual(t, "batchJobs", jobs, []ProwJob{
+		{Type: "batch", Repo: "a", State: "success", Refs: "1", Context: "$"},
+		{Type: "batch", Repo: "a", State: "success", Refs: "2", Context: "!"},
+	})
+}
+
+func TestBatchRefToBatch(t *testing.T) {
+	_, strconvErr := strconv.ParseInt("a", 10, 32)
+	for _, test := range []struct {
+		ref      string
+		expected Batch
+		err      error
+	}{
+		{"m:a", Batch{"m", "a", nil}, nil},
+		{"m:a,1:b", Batch{"m", "a", []BatchPull{{1, "b"}}}, nil},
+		{"m:a,1:b,2:c", Batch{"m", "a", []BatchPull{{1, "b"}, {2, "c"}}}, nil},
+		{"asdf", Batch{}, errors.New("bad batchref")},
+		{"m:a,a:3", Batch{}, strconvErr},
+	} {
+		batch, err := batchRefToBatch(test.ref)
+		expectEqual(t, "error", err, test.err)
```

Comment:
It's bad to do this. If you want to test that the error is the proper type then define a new type for it and check that a type assertion works.

This test breaks if you change the text of the error which is annoying. Not a big deal, just annoying.